src/main/ruby/truffleruby/core/kernel.rb

```diff
@@ -652,7 +652,12 @@ def warn(*messages, uplevel: undefined)
         end
         $stderr.write message
       else
-        Warning.warn(message)
+        category = if Primitive.undefined?(category)
+                     nil
+                   else
+                     Truffle::Type.rb_convert_type(category, Symbol, :to_sym)
+                   end
+        Warning.warn(message, category: category)
```

Comment:
We shouldn't pass a `category` keyword if `category` is `nil`, in order to improve compatibility with `Warning.warn` not handling kwargs.

This would be good to spec (add one under `spec/ruby`).

Reply:
I think we have one, but I was a bit confused by it: https://github.com/Strech/truffleruby/blob/bfed12342e147d14bb19f0a6fdc00a1b5706731a/spec/ruby/core/warning/warn_spec.rb#L56-L65

Reply:
Interesting, I didn't expect that behavior (seems less compatible) but since that's the case, good as-is

Reply:
It's more complicated, it looks like CRuby checks if Warning.method(:warn) accepts kwargs and if only if it does it passes `category: value` to it.